lib/wallet/wallet.js

```diff
@@ -1136,6 +1136,238 @@ class Wallet extends EventEmitter {
   }
 
+  /**
+   * Fill a transaction with as many of passed coins
+   * as possible
+   * @see MTX#selectCoins
+   * @see MTX#fill
+   * @param {MTX} mtx - _Must_ be a mutable transaction.
+   * @param {Object?} options
+   * @param {(String|Number)?} options.account - If no account is
+   * specified, coins from the entire wallet will be filled.
+   * @param {Boolean} options.round - Whether to round to the nearest
+   * kilobyte for fee calculation.
+   * See {@link TX#getMinFee} vs. {@link TX#getRoundFee}.
+   * @param {Rate} options.rate - Rate used for fee calculation.
+   * @param {Boolean} options.confirmed - Select only confirmed coins.
+   * @param {Boolean} options.free - Do not apply a fee if the
+   * transaction priority is high enough to be considered free.
+   * @param {Amount?} options.hardFee - Use a hard fee rather than
+   * calculating one.
+   * @param {Number|Boolean} options.subtractFee - Whether to subtract the
+   * fee from existing outputs rather than adding more inputs.
+   */
+
+  async fill(mtx, coins, options, force) {
+    const unlock = await this.fundLock.lock(force);
+    try {
+      return await this._fill(mtx, coins, options);
+    } finally {
+      unlock();
+    }
+  }
+
+  /**
+   * Fill a transaction with as many of passed coins
+   * as possible
+   * @private
+   * @see MTX#selectCoins
+   * @see MTX#fill
+   * @param {MTX} mtx - _Must_ be a mutable transaction.
+   * @param {Object?} options
+   * @param {(String|Number)?} options.account - If no account is
+   * specified, coins from the default account will be used to fill.
+   * Pass -1 to fill from the entire wallet.
+   * @param {String?} options.selection - Coin selection priority. Can
+   * be `age`, `random`, or `all`. (default=age).
+   * @param {Boolean} options.round - Whether to round to the nearest
+   * kilobyte for fee calculation.
+   * See {@link TX#getMinFee} vs. {@link TX#getRoundFee}.
+   * @param {Rate} options.rate - Rate used for fee calculation.
+   * @param {Boolean} options.confirmed - Select only confirmed coins.
+   * @param {Boolean} options.free - Do not apply a fee if the
+   * transaction priority is high enough to be considered free.
+   * @param {Amount?} options.hardFee - Use a hard fee rather than
+   * calculating one.
+   * @param {Number|Boolean} options.subtractFee - Whether to subtract the
+   * fee from existing outputs rather than adding more inputs.
+   */
+
+  async _fill(mtx, coins, options) {
+    let accountIndex = 0;
+    if (options.account) {
+      assert(
+        typeof options.account === 'number',
+        'Account option should be a number corresponding to index of account'
+      );
+      accountIndex = options.account;
+    }
+
+    // pick random address from options.addresses or generate new one
+    const addressCount = options.addresses ? options.addresses.length : 0;
+    let address;
+    if (addressCount) {
+      const index =
+        Math.floor(Math.random() * (addressCount + 1));
```

Comment:
If only one address defined in adresses array, addressCount = 1, this would occasionally result in index = 1, resulting in `undefined` address.

Reply:
Good catch. Careless bug. Would also have problems if the random selector picked something at the end of the list. Should be fixed and am testing with a single address in the array in the rpc test now too.